datafusion/core/src/physical_optimizer/join_filter_pushdown.rs

```diff
@@ -0,0 +1,112 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Pushdown the dynamic join filters down to scan execution if there is any
+
+use std::sync::Arc;
+
+#[cfg(feature = "parquet")]
+use crate::datasource::physical_plan::ParquetExec;
+use crate::physical_plan::ExecutionPlan;
+use crate::{config::ConfigOptions, error::Result, physical_plan::joins::HashJoinExec};
+use datafusion_common::tree_node::{Transformed, TransformedResult};
+use datafusion_physical_optimizer::PhysicalOptimizerRule;
+use datafusion_physical_plan::joins::DynamicFilterInfo;
+
+/// this rule used for pushing the build side statistic down to probe phase
+#[derive(Default, Debug)]
+pub struct JoinFilterPushdown {}
+
+impl JoinFilterPushdown {
+    #[allow(missing_docs)]
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl PhysicalOptimizerRule for JoinFilterPushdown {
```

Comment:
This is executed during plan time, won't it be reasonable to execute this during execution of `HashJoin` after build side is loaded?